File: 223_RectangleArea.c
//20.41%
int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
    //Check boundary conditions
	if(C < A || D < B || G < E || H < F)
		return 0;
	
	int area1 = (C - A) * (D - B);
	int area2 = (G - E) * (H - F);
	int commonArea = 0;
	int h = 0, w = 0; //h : commonHeight, w : commonWidth
	if( C <= E || G <= A || B >= H || F >= D){
		commonArea = 0;
		return area1 + area2;
	}
	if(E > A){
		if(G < C)
			w = G - E;
		else
			w = C - E;
	}else{
		if(G < C)
			w = G - A;
		else
			w = C - A;
	}
	
	if(B > F){
		if(H < D)
			h = H - B;
		else
			h = D - B;
	}else{
		if(H < D)
			h = H - F;
		else
			h = D - F;
	}
	commonArea = h * w;
	return area1 - commonArea + area2;
		
}

File: 223_RectangleArea.py
# 50.64%
class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """
        if A > C or B > D or E > G or F > H:
            return 0
        
        area1, area2 = (C - A) * (D - B), (G - E) * (H - F)
        if C <= E or G <= A or H <= B or D <= F:
            return area1 + area2
        C, A = C if C < G else G, A if A > E else E
        D, B = D if D < H else H, B if B > F else F
        return area1 - (C - A) * (D - B) + area2
		
def computeArea(self, A, B, C, D, E, F, G, H):
    overlap = max(min(C,G)-max(A,E), 0)*max(min(D,H)-max(B,F), 0)
    return (A-C)*(B-D) + (E-G)*(F-H) - overlap